Modules/src/StatusAggregator.cc

```diff
@@ -116,7 +116,8 @@ namespace ChimeraTK {
   int StatusAggregator::getPriority(StatusOutput::Status status) {
     using Status = StatusOutput::Status;
 
-    const std::map<PriorityMode, std::map<Status, int32_t>> map_priorities{
+    // static helps against initializing over and over again
+    static const std::map<PriorityMode, std::map<Status, int32_t>> map_priorities{
```

Comment:
Can you work with constexpr. Maybe this improves performance even more.

Reply:
Or a single map with a pair as key?

Reply:
constexpr does not work here. It would require that std::map has a constexpr constructor.
A single map with key would be better, agreed.
However, if we really want to make this fast, the solution would be defining a 2d array  like
int map_priorities[enumValuesOfPriorityMode][enumValuesOfStatusCode]
and use that instead of the map.
But for that, we would have hard-coded knowledge about enum values, which I also don't like...

Reply:
I would leave it as it is until we really need a better optimization. The idea with key=pair(mode, status) would work but inflate initialization code for a very minor improvement.

Reply:
You could try replacing the std::map with a boost::hana::map, which supports constexpr, if you want to optimise that. Might be pretty straight forward.

https://www.boost.org/doc/libs/1_71_0/libs/hana/doc/html/structboost_1_1hana_1_1map.html